1.2/sd-client/src/main/java/ServiceDirectoryClient.java

```diff
@@ -0,0 +1,256 @@
+/**
+ * Copyright (c) 2013-2014 by Cisco Systems, Inc.
+ * All rights reserved.
+ */
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.StringTokenizer;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import com.cisco.oss.foundation.directory.ServiceDirectory;
+import com.cisco.oss.foundation.directory.client.DirectoryServiceRestfulClient;
+import com.cisco.oss.foundation.directory.entity.ServiceInstance;
+import com.cisco.oss.foundation.directory.exception.ServiceException;
+
+/**
+ * The ServiceDirectory Tool to operate ServiceInstance in the remote Directory Server.
+ */
+public class ServiceDirectoryClient {
+
+    private static Options options = new Options();
+
+    static {
+        Option help = Option.builder("h").longOpt("help").desc("print this message").
+                build(); //boolean option
+        Option server = Option.builder("s").longOpt("server").required(false).numberOfArgs(1).desc(
```

Comment:
it depends, if not --server option provided, vcsdirsvc:2013 used by default. I think it ok because we can always provide the -server option in our script. It doesn't harm by keeping a simpler life when in some case we need to type input from keyboard.